benchmarks/gcdx.jl

```diff
@@ -0,0 +1,62 @@
+function gcdx_bigint(a::fmpz, b::fmpz)
+  g, s, t = gcdx(BigInt(a), BigInt(b))
+  return fmpz(g), fmpz(s), fmpz(t)
+end
+
+function gcdx_fmpz(a::fmpz, b::fmpz)
+  d = ZZ()
+  x = ZZ()
+  y = ZZ()
+  ccall((:fmpz_xgcd_canonical_bezout, :libflint), Nothing,
+        (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), d, x, y, a, b)
+  return d, x, y
+end
+
+function run_gcdx_bigint(x::Vector{fmpz}, y::Vector{fmpz})
+  for ix in x, iy in y
+    gcdx_bigint(ix, iy)
+  end
+end
+    
+function run_gcdx_fmpz(x::Vector{fmpz}, y::Vector{fmpz})
+  for ix in x, iy in y
+    gcdx_fmpz(ix, iy)
+  end
+end
+
+function benchmark_gcdx()
+  print("benchmark_gcdx ...\n")
+
+  # small size
+  sp = Random.Sampler(Random.MersenneTwister(),
+                      ZZ(0):ZZ(2)^(Sys.WORD_SIZE - 2) - 1)
+  x = [rand(sp) for _ in 1:100]
+  y = [rand(sp) for _ in 1:100]
```

Comment:
Just as a note: it's typical to explicitly create a `Sampler` when random generation must be as fast as possible. But here, random generation is not included in the benchmark, so you could as well have assigned only the range to a variable `r` and then call `rand(r)`. In particular, here you just call `rand(sp)`, which is "slow" on Julia versions 1.3-1.6, for fast code you would want `rand(rng, sp)` (and note that although it won't matter here, a `Sampler` `sp` is created for a specific RNG type `T`, and `rand(rng, sp)` should be called with `rng::T`; but in Julia 1.7, the implicit RNG won't be `MersenneTwister` anymore).

Reply:
Thanks!